tests/test_file_utils.py

```diff
@@ -156,3 +157,36 @@ def test_find_first_smaller_chunksize_than_pattern_doesnt_hang():
     fake_file = io.BytesIO(b"0123456789_pattern")
     with pytest.raises(ValueError):
         find_first(fake_file, b"pattern", chunk_size=0x5)
+
+
+@pytest.mark.parametrize(
+    "content, start_offset, size, buffer_size, expected",
+    (
+        pytest.param(b"", 0, 1, 10, [], id="empty_file"),
+        pytest.param(b"0123456789", 0, 6, 3, [b"012", b"345"], id="from_start"),
+        pytest.param(b"0123456789", 0, 0, 3, [], id="zero_size"),
+        pytest.param(b"0123456789", 0, 5, 3, [b"012", b"34"], id="size_buffersize"),
+        pytest.param(b"0123456789", 0, 5, 100, [b"01234"], id="big_buffersize"),
+        pytest.param(b"0123456789", 0, 10, 10, [b"0123456789"], id="real_all"),
+        pytest.param(b"0123456789", 0, 10, 100, [b"0123456789"], id="big_buffersize"),
+        pytest.param(b"0123456789", 0, 100, 10, [b"0123456789"], id="big_size"),
+        pytest.param(b"0123456789", 5, 3, 2, [b"56", b"7"], id="middle_offset"),
+        pytest.param(b"0123456789", 10, 3, 2, [], id="offset_bigger_than_file"),
```

Comment:
Shouldn't this be a ValueError too?

Reply:
Technically, it is possible to `seek` a bigger offset and do a `read` - it would just return empty bytes, which would stop the file iteration. So I thought it would be reasonable to keep this ("default") behavior.

Reply:
Wow:
```python
>>> from pathlib import Path
>>> path.stat().st_size
744
>>> fp = path.open("rb")
>>> fp.seek(1000)
1000
>>> fp.tell()
1000
>>> fp.read()
b''
```
Ok.